core/templates/services/auth.service.ts

```diff
@@ -60,58 +147,25 @@ export class AuthService {
       ['localhost', 9099] : undefined;
   }
 
-  /**
-   * Prompts the user to sign-in with a trusted identity provider, then fulfills
-   * with true if the user is new, otherwise fulfills with false.
-   * Rejects when sign-in failed.
-   */
-  async signInAsync(): Promise<boolean> {
-    if (!this.angularFireAuth) {
-      throw new Error('AngularFireAuth is not available');
+  async handleRedirectResultAsync(): Promise<void> {
+    const creds = await this.authServiceImpl.getRedirectResultAsync();
+    if (creds.user) {
+      const idToken = await creds.user.getIdToken();
+      return this.authBackendApiService.beginSessionAsync(idToken);
+    } else {
+      return Promise.reject(null);
     }
-
-    const creds = AuthService.firebaseEmulatorIsEnabled ?
-      await this.emulatorSignInAsync() : await this.productionSignInAsync();
-
-    const idToken = await creds.user.getIdToken();
-    await this.authBackendApiService.beginSessionAsync(idToken);
-
-    return creds.additionalUserInfo.isNewUser;
   }
 
-  async signOutAsync(): Promise<void> {
-    if (!this.angularFireAuth) {
-      throw new Error('AngularFireAuth is not available');
-    }
-
-    await this.angularFireAuth.signOut();
+  async signInWithRedirectAsync(): Promise<void> {
+    return this.authServiceImpl.signInWithRedirectAsync();
   }
 
-  /** Assumes that angularFireAuth is not null. */
-  private async emulatorSignInAsync(): Promise<firebase.auth.UserCredential> {
-    const email = prompt('Please enter the email address to sign-in with');
-    const password = await md5(email);
-    try {
-      return await this.angularFireAuth.signInWithEmailAndPassword(
-        email, password);
-    } catch (err) {
-      if (err.code !== 'auth/user-not-found') {
-        throw err;
-      }
-    }
-    return await this.angularFireAuth.createUserWithEmailAndPassword(
-      email, password);
-  }
-
-  /** Assumes that angularFireAuth is not null. */
-  private async productionSignInAsync(): Promise<firebase.auth.UserCredential> {
-    const provider = new firebase.auth.GoogleAuthProvider();
-    // Oppia only needs an email address for account management.
-    provider.addScope('email');
-    // Always prompt the user to select an account, even when they only own one.
-    provider.setCustomParameters({prompt: 'select_account'});
-    await this.angularFireAuth.signInWithRedirect(provider);
-    return await this.angularFireAuth.getRedirectResult();
+  async signOutAsync(): Promise<void> {
+    await Promise.all([
+      this.authServiceImpl.signOutAsync(),
```

Comment:
Is this calling into Firebase? Also, are there any potential issues with doing these simultaneously?